src/main/java/io/nats/client/impl/NatsJetStreamPullSubscription.java

```diff
@@ -149,7 +150,7 @@ private List<Message> _fetch(int batchSize, long maxWaitMillis) {
                     batchLeft--;
                 }
                 // try again while we have time
-                timeLeft = maxWaitMillis - (System.currentTimeMillis() - start);
+                timeLeft = maxWaitMillis - TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
```

Comment:
What do you think about making everything nanos?
```
long maxWaitNanos = maxWaitMillis * 1_000_000; // is million right?
...
long timeLeft = maxWaitNanos;
...
Message msg = nextMessageInternal( Duration.ofNanos(timeLeft) );
...
timeLeft = maxWaitNanos - (System.nanoTime() - start);
```

Also, I really need you to verify your commits. It's an CNCF rule, I'm allowed to skip it on occasion, but I'd prefer not to.

Reply:
Ah gotcha. I was just using the github workspace to make these changes. Let me look into how to do it. I'll ask for the review once again after I figure it out.

Reply:
Yeah, nanos would be better here too, since the `poll` method we use in MessageQueue uses nanos.